insights/combiners/sysctl_conf.py

```diff
@@ -0,0 +1,56 @@
+"""
+Sysctl configuration files
+==========================
+"""
+import re
+
+from insights.core.plugins import combiner
+from insights.parsers.sysctl import SysctlDConf
+
+
+@combiner(SysctlDConf)
```

Comment:
Hi @ryan-blakley, I think it would be better to just add a `SysctlConf` combiner that combines both the `SysctlConf` parser and `SysctlDConf` parsers, and take the `SysctlDConf` parsers in the high priority (take their values as the final values). See the [sysctl-man-page:CONFIGURATION DIRECTORIES AND PRECEDENCE](https://man7.org/linux/man-pages/man5/sysctl.d.5.html)  part.

And I think all the existing rules should also be updated to use the new combiner once it's merged. What do you think?

Reply:
Hey @xiangce so I contemplated adding the `SysctlConf` parser to the combiner also. But I decided not to, because technically `/etc/sysctl.conf` only gets read because the `/etc/sysctl.d/99-sysctl.conf` file is a symlink to it. If that symlink is missing `/etc/sysctl.conf` doesn't get read in. I guess technically we could deprecate the `SysctlConf` parser for the combiner.

Based on that man page link, do you think I need to add a spec for the `/usr/lib/sysctl.d/*.conf` files also, since a third party package could add sysctls via that path? I see there are conf files on my minimal RHEL 7 test system in that directory.

Thank you for mentioning the order part, I will need to add some logic to update the dictionary based on the file name order.

Yes I think all of the rules that use the `SysctlConf` parser should be updated to use the new combiner once merged.

Reply:
Hey @ryan-blakley - Thanks for the sharing. 
1. Regarding the `/etc/sysctl.d/99-sysctl.conf`. Although it's a fact in RHEL 7/8/(and 9) that the `/etc/sysctl.d/99-sysctl.conf` will be created and point to the `/etc/sysctl.conf` by default, it should not be a blocker for the combiner to take it as a source. Since it's a configuration file, it's possible for the users (customers) to remove the symbol link file intentionally or unintentionally, which results in rules' false positive in the specific situation.
And I didn't find this symbol link file in the minimal RHEL 6.10 system which is still supported by Red Hat. So I think we'd better add it to the combiner, after all, even if it's a symbol link to the `/etc/sysctl.conf`, the combiner's result won't be wrong. 
2. I agree with you that we should add a spec for the `/usr/lib/sysctl.d/*.conf` files together and add its parser to the combiner.

Thanks.

Reply:
Ahhh I didn't see that on RHEL 6 that it reads /etc/sysctl.conf, I added it to the combiner. I also added the /usr/lib/sysctl.d spec, parser, and added it to the combiner.